TaskLayer/MbrAnalysis/MbrSpectralMatch.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using EngineLayer;
+using FlashLFQ;
+
+namespace TaskLayer.MbrAnalysis
+{
+    public class MbrSpectralMatch
+    {
+        public PeptideSpectralMatch spectralLibraryMatch { get; set; }
+
+        public ChromatographicPeak acceptorPeak { get; set; }
+
+        public bool originalMatchFound { get; private set; }
+
+        public PeptideSpectralMatch originalSpectralMatch { get; private set; }
+
+        public MbrSpectralMatch(PeptideSpectralMatch spectralLibraryMatch, ChromatographicPeak acceptorPeak)
+        {
+            this.spectralLibraryMatch = spectralLibraryMatch;
+            this.acceptorPeak = acceptorPeak;
+            originalMatchFound = false;
+        }
+
+        public void FindOriginalPsm(List<PeptideSpectralMatch> originalSearchPsms)
+        {
+            if (spectralLibraryMatch == null) return;
+            int oneBasedScanNumber = spectralLibraryMatch.ScanNumber;
+            string spectraFile = spectralLibraryMatch.FullFilePath;
+            string fullSequence = spectralLibraryMatch.FullSequence;
+
+            PeptideSpectralMatch originalPsm = originalSearchPsms
+                .Where(p => p.FullFilePath.Equals(spectraFile))
+                .Where(p => p.ScanNumber == oneBasedScanNumber)
+                .Where(p => p.FullSequence == fullSequence)
+                .FirstOrDefault();
+            if (originalPsm != null)
```

Comment:
Do you want to take into account ambiguous PSMs here or only work with Level 1 ID's?

Reply:
Only level 1 IDs. How should I go about that?

Reply:
.Where(p => p.FullSequence == fullSequence || p.FullSequence.Split('|').Any(m => m == fullSequence))

Reply:
done, thanks